### PR TITLE
Automated cherry pick of #123809: fix stateful set pod recreation and event spam

### PR DESCRIPTION
**Pull request by atiratree**
Cherry pick of #123809 on release-1.30.

#123809: fix stateful set pod recreation and event spam

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by atiratree** [comment]
/hold 
TestStatefulSetControl is flaking

```
{Failed;Failed;  === RUN   TestStatefulSetControl/CreatePodFailure/Burst/nullPolicy/StatefulSetAutoDeletePVCEnabled
W0426 13:54:22.876820   54128 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0426 13:54:22.876866   54128 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0426 13:54:22.876906   54128 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0426 13:54:22.876968   54128 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
    stateful_set_control_test.go:526: StatefulSetControl did not return InternalError, found %!s(<nil>)
    --- FAIL: TestStatefulSetControl/CreatePodFailure/Burst/nullPolicy/StatefulSetAutoDeletePVCEnabled (0.04s)
;=== RUN   TestStatefulSetControl
--- FAIL: TestStatefulSetControl (7.75s)
;}
```

**Review by soltysh — APPROVED** [review]
/approve

**Comment by atiratree** [comment]
unrelated
/retest

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Review by Verolop — COMMENTED** [review]
/lgtm 
(release managers)

**Pull request merged**
